### ISSUE TITLE
Progress indication for commands like `create` and Long Running Operations

### ISSUE THREAD

**Issue by iainfoulds** [Azure/azure-cli#987]
Didn't see an existing issue for this. Aplogies if so.

When issuing a command, such as `az vm create`, you get no indication of progress. The CLI just sits until the entire operation has completed. Looking in the portal, you can see as the storage, then networking, then VM, etc., is created. node.js CLI shows indication by default as each resource is in progress and completed, etc.

az --version
azure-cli (2016.09.05.nightly)

**Comment by lostintangent** [comment]
Creating a VM and just waiting for a couple minutes definitely "feels" a little odd, especially after being accustomed to how many other CLIs behave (e.g. doing a Docker build, performing a Git deploy to Heroku and seeing the buildpack process), and expecting some sort of visual indication of progress.

**Comment by johanste** [comment]
Would providing a better (actually working) --verbose option solve the problem? We have tried really, really (maybe too) hard to follow the "silence is golden, only generate error/non "return value" output for unexpected events/errors.

**Comment by iainfoulds** [comment]
Taking the VM creation example, since that's probably a pretty well used command, there are a number of other resources it's validating or creating. So it would nice to at least see, "yep, storage found", "yep, vnet and nic found", "now creating VM", etc. Especially since you're waiting on response from the underlying APIs each time. Maybe this goes under a more readable output from `--verbose`.

If I'm just creating a new VNet, there's probably not a lot I'm waiting on, and it should be a pretty quick operation. I don't think anything needs to be displayed until the operation has completed. Same with creating a new storage account, or a new public IP address, etc. 

But on those longer-running operations, like creating a VM or deploying from a template, it would be nice to get an indication rather than nothing really being displayed for minutes until the whole operation has completed. Then you can at least see, "OK, all the supporting resources are in place, now it's actually churning on spinning up the new VM"


Actually, on the template deployment, bonus points if that then notes the deployment has been handed off, and is in progress - the existing node CLI returns you to the prompt within a few seconds of starting a new deployment (`azure group create --template-uri azuredeploy.json`) and says `Provisioning State: Succeeded` when really that's acknowledging the platform accepted the deployment task, not having finished creating all the actual resources within the template :) 

Double bonus points for being able to then track that deployment progress. But we're getting outside the scope of this issue and is handled elsewhere, really!

**Comment by colemickens** [comment]
I am interested in this for CI scenarios.

I can't use `--verbose` there because a goal is to get this to be something I can expose publicly and there are too many secrets in the verbose output right now for me to use that as a status indicator.

Particularly for an ARM template deployment that can take 1 minutes to 20 minutes, some status would be nice (if the user asks for it explicitly... I like the silence-on-success pattern).

Something like a summary output that refereshed, or maybe just appended the resource name from the ARM template as soon as it moves into Success provisioning state?

**Comment by spboyer** [comment]
Silence is like golfing in the dark. I agree with @colemickens on the `--verbose` flag for the amount of information. 

> Particularly for an ARM template deployment that can take 1 minutes to 20 minutes, some status would be nice (if the user asks for it explicitly... I like the silence-on-success pattern).

Some middle ground must be attainable. 

```bash
Performing {Task} [insert spinner thigy]
Result: Success/Fail/Warning 
Message:
```
Move on to the next task.

**Comment by tjprescott** [comment]
#3510 aims to resolve this.
